### DISCUSSION TITLE
Interfaces in custom types for action handlers

### DISCUSSION THREAD

**Discussion by charklewis** [hasura/graphql-engine#8510] [General] [answered]
I saw on the Hasura docs is says `It is currently not possible to define Interfaces and Union types as custom types`. Are there any plans to support this? About half of my action handlers would be using this feature and at the moment I have to duplicate the types (which isn't scaling very well).

**Comment by seanonthenet** [comment, marked as answer]
There is a issue out for union (and interface) types: https://github.com/hasura/graphql-engine/issues/2311

And a related one about allowing the use of existing types: https://github.com/hasura/graphql-engine/issues/5001

I also hope these get prioritised.

**Reply by dsandip** [reply]
@charklewis Could you please upvote/+1 the issues shared by @seanonthenet so help prioritise these issues higher?
